### PR TITLE
Support active state for non-readonly custom buttons inside readonly editor (T905970)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by ksercs — DISMISSED** [review]

File: js/ui/text_box/texteditor_button_collection/custom.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ export default class CustomButton extends TextEditorButton {
         this._addToContainer($element);
 
         const instance = editor._createComponent($element, Button, extend({}, this.options, {
+            parentIndependentState: true,
```
Comment (DokaRus) [inline js/ui/text_box/texteditor_button_collection/custom.js]: Let's pick up another name for the option

Comment (alexander-kotov-dx) [reply js/ui/text_box/texteditor_button_collection/custom.js]: Renamed to ignoreParentReadOnly

**Review by ksercs — DISMISSED** [review]

File: js/ui/widget/ui.widget.js [hunk 2]
```diff
@@ -148,6 +149,7 @@ const Widget = DOMComponent.inherit({
         this._renderFocusState();
         this._attachFeedbackEvents();
         this._attachHoverEvents();
+        this._addStateClasses();
```
Comment (DokaRus) [inline js/ui/widget/ui.widget.js]:
```suggestion
        this._toggleIndependentState();
```

File: js/ui/widget/ui.widget.js [hunk 3]
```diff
@@ -460,6 +462,10 @@ const Widget = DOMComponent.inherit({
         this.setAria('disabled', value || undefined);
     },
 
+    _addStateClasses() {
```
Comment (DokaRus) [inline js/ui/widget/ui.widget.js]:
```suggestion
    _toggleIndependentState() {
```

File: js/ui/widget/ui.widget.js [hunk 4]
```diff
@@ -495,6 +501,9 @@ const Widget = DOMComponent.inherit({
             case 'hint':
                 this._renderHint();
                 break;
+            case 'ignoreParentReadOnly':
+                this._addStateClasses();
```
Comment (DokaRus) [inline js/ui/widget/ui.widget.js]:
```suggestion
                this._toggleIndependentState();
```

**Review by DokaRus — APPROVED** [review]

**Pull request merged**